src/Mutex/DistributedMutex.php

```diff
@@ -34,6 +34,9 @@ class DistributedMutex extends AbstractSpinlockWithTokenMutex implements LoggerA
     public function __construct(array $mutexes, float $acquireTimeout = 3, float $expireTimeout = \INF)
     {
         parent::__construct('', $acquireTimeout, $expireTimeout);
+        \Closure::bind(function () {
+            $this->key = 'distributed';
```

Comment:
@mvorisek Why did you decide to pin the key to this value?

This key is now used for the lock across all underlying mutex clients, overriding the key provided to each client in their constructor. Is this intentional?

Would it not make more sense to do the following?

* Loop through `array $mutexes` to ensure their keys are all exactly the same.
* Then, use that key as the `DistributedMutex` key instead of a fixed string `distributed`.

Reply:
Does the key really matter, where exactly it is used?

> * Loop through `array $mutexes` to ensure their keys are all exactly the same.

Does requiring the keys to be the same provide any safety advantage?

In anycase, improvement pull requests are welcomed.